# 📊 Data Understanding

## Project Information

- **Project:** Industry Simulation Program – Data Analyst
- **Week:** 2
- **Day:** 3
- **Topic:** Data Understanding
- **Dataset:** 100000 Sales Records

---

## Objectives

The objectives of this notebook are:

- Load the dataset using Pandas.
- Understand the dataset structure.
- Identify the number of rows and columns.
- Review column names and data types.
- Generate initial descriptive statistics.

# 1. Import Library

In [3]:
import pandas as pd

# 2. Load Dataset

In [5]:
import os

os.listdir("/content")

['.config', 'sample_data']

In [6]:
from google.colab import files

uploaded = files.upload()

Saving 100000 Sales Records.csv to 100000 Sales Records.csv


# 3. Preview Dataset

The first five rows are displayed to understand the structure of the dataset and verify that it has been loaded correctly.

In [9]:
df.head()

,Region,Country,Item Type,Sales Channel,Order Priority,Order Date,Order ID,Ship Date,Units Sold,Unit Price,Unit Cost,Total Revenue,Total Cost,Total Profit
0,Middle East and North Africa,Azerbaijan,Snacks,Online,C,10/8/2014,535113847,10/23/2014,934,152.58,97.44,142509.72,91008.96,51500.76
1,Central America and the Caribbean,Panama,Cosmetics,Offline,L,2/22/2015,874708545,2/27/2015,4551,437.20,263.33,1989697.20,1198414.83,791282.37
2,Sub-Saharan Africa,Sao Tome and Principe,Fruits,Offline,M,12/9/2015,854349935,1/18/2016,9986,9.33,6.92,93169.38,69103.12,24066.26
3,Sub-Saharan Africa,Sao Tome and Principe,Personal Care,Online,M,9/17/2014,892836844,10/12/2014,9118,81.73,56.67,745214.14,516717.06,228497.08
4,Central America and the Caribbean,Belize,Household,Offline,H,2/4/2010,129280602,3/5/2010,5858,668.27,502.54,3914725.66,2943879.32,970846.34



  
    

    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }


  
    
      
      Region
      Country
      Item Type
      Sales Channel
      Order Priority
      Order Date
      Order ID
      Ship Date
      Units Sold
      Unit Price
      Unit Cost
      Total Revenue
      Total Cost
      Total Profit
    
  
  
    
      0
      Middle East and North Africa
      Azerbaijan
      Snacks
      Online
      C
      10/8/2014
      535113847
      10/23/2014
      934
      152.58
      97.44
      142509.72
      91008.96
      51500.76
    
    
      1
      Central America and the Caribbean
      Panama
      Cosmetics
      Offline
      L
      2/22/2015
      874708545
      2/27/2015
      4551
      437.20
      263.33
      1989697.20
      1198414.83
      791282.37
    
    
      2
      Sub-Saharan Africa
      Sao Tome and Principe
      Fruits
      Offline
      M
      12/9/2015
      854349935
      1/18/2016
      9986
      9.33
      6.92
      93169.38
      69103.12
      24066.26
    
    
      3
      Sub-Saharan Africa
      Sao Tome and Principe
      Personal Care
      Online
      M
      9/17/2014
      892836844
      10/12/2014
      9118
      81.73
      56.67
      745214.14
      516717.06
      228497.08
    
    
      4
      Central America and the Caribbean
      Belize
      Household
      Offline
      H
      2/4/2010
      129280602
      3/5/2010
      5858
      668.27
      502.54
      3914725.66
      2943879.32
      970846.34
    
  


    

  
    

  
    
  
    

  
    .colab-df-container {
      display:flex;
      gap: 12px;
    }

    .colab-df-convert {
      background-color: #E8F0FE;
      border: none;
      border-radius: 50%;
      cursor: pointer;
      display: none;
      fill: #1967D2;
      height: 32px;
      padding: 0 0 0 0;
      width: 32px;
    }

    .colab-df-convert:hover {
      background-color: #E2EBFA;
      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);
      fill: #174EA6;
    }

    .colab-df-buttons div {
      margin-bottom: 4px;
    }

    [theme=dark] .colab-df-convert {
      background-color: #3B4455;
      fill: #D2E3FC;
    }

    [theme=dark] .colab-df-convert:hover {
      background-color: #434B5C;
      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);
      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));
      fill: #FFFFFF;
    }
  

    
      const buttonEl =
        document.querySelector('#df-1eac2f72-08f6-4fc8-81f5-6d3ceee7118c button.colab-df-convert');
      buttonEl.style.display =
        google.colab.kernel.accessAllowed ? 'block' : 'none';

      async function convertToInteractive(key) {
        const element = document.querySelector('#df-1eac2f72-08f6-4fc8-81f5-6d3ceee7118c');
        const dataTable =
          await google.colab.kernel.invokeFunction('convertToInteractive',
                                                    [key], {});
        if (!dataTable) return;

        const docLinkHtml = 'Like what you see? Visit the ' +
          '<a target="_blank" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'
          + ' to learn more about interactive tables.';
        element.innerHTML = '';
        dataTable['output_type'] = 'display_data';
        await google.colab.output.renderOutput(dataTable, element);
        const docLink = document.createElement('div');
        docLink.innerHTML = docLinkHtml;
        element.appendChild(docLink);
      }
    
  


    
  


In [10]:
df.shape

(100000, 14)

# 5. Column Names

Display all column names in the dataset.

In [11]:
df.columns

Index(['Region', 'Country', 'Item Type', 'Sales Channel', 'Order Priority',
       'Order Date', 'Order ID', 'Ship Date', 'Units Sold', 'Unit Price',
       'Unit Cost', 'Total Revenue', 'Total Cost', 'Total Profit'],
      dtype='object')

# 6. Dataset Information

Display general information about the dataset including data types, non-null values, and memory usage.

In [12]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 100000 entries, 0 to 99999
Data columns (total 14 columns):
 #   Column          Non-Null Count   Dtype  
---  ------          --------------   -----  
 0   Region          100000 non-null  object 
 1   Country         100000 non-null  object 
 2   Item Type       100000 non-null  object 
 3   Sales Channel   100000 non-null  object 
 4   Order Priority  100000 non-null  object 
 5   Order Date      100000 non-null  object 
 6   Order ID        100000 non-null  int64  
 7   Ship Date       100000 non-null  object 
 8   Units Sold      100000 non-null  int64  
 9   Unit Price      100000 non-null  float64
 10  Unit Cost       100000 non-null  float64
 11  Total Revenue   100000 non-null  float64
 12  Total Cost      100000 non-null  float64
 13  Total Profit    100000 non-null  float64
dtypes: float64(5), int64(2), object(7)
memory usage: 10.7+ MB


# 7. Descriptive Statistics

Generate summary statistics for numerical columns.

In [13]:
df.describe()

,Order ID,Units Sold,Unit Price,Unit Cost,Total Revenue,Total Cost,Total Profit
count,1.000000e+05,100000.000000,100000.000000,100000.000000,1.000000e+05,1.000000e+05,1.000000e+05
mean,5.503956e+08,5001.446170,266.703989,188.019711,1.336067e+06,9.419755e+05,3.940912e+05
std,2.593219e+08,2884.575424,216.940081,175.706023,1.471768e+06,1.151828e+06,3.795986e+05
min,1.000089e+08,1.000000,9.330000,6.920000,1.866000e+01,1.384000e+01,4.820000e+00
25%,3.260464e+08,2505.000000,109.280000,56.670000,2.797533e+05,1.629283e+05,9.590000e+04
50%,5.477185e+08,5007.000000,205.700000,117.110000,7.898916e+05,4.679374e+05,2.836575e+05
75%,7.750785e+08,7495.250000,437.200000,364.690000,1.836490e+06,1.209475e+06,5.683841e+05
max,9.999965e+08,10000.000000,668.270000,524.960000,6.682700e+06,5.249075e+06,1.738700e+06


# 8. Random Sample

Display random rows from the dataset to inspect data consistency.

In [14]:
df.sample(5)

,Region,Country,Item Type,Sales Channel,Order Priority,Order Date,Order ID,Ship Date,Units Sold,Unit Price,Unit Cost,Total Revenue,Total Cost,Total Profit
89612,Sub-Saharan Africa,Botswana,Personal Care,Online,H,12/8/2013,465026223,1/16/2014,448,81.73,56.67,36615.04,25388.16,11226.88
27801,Australia and Oceania,Tuvalu,Snacks,Online,C,4/29/2015,919268405,5/29/2015,804,152.58,97.44,122674.32,78341.76,44332.56
18429,Middle East and North Africa,Tunisia,Fruits,Offline,C,9/9/2012,390594279,9/21/2012,4382,9.33,6.92,40884.06,30323.44,10560.62
37260,Sub-Saharan Africa,Eritrea,Household,Online,M,6/2/2017,291387307,7/12/2017,6834,668.27,502.54,4566957.18,3434358.36,1132598.82
74219,Sub-Saharan Africa,Seychelles,Clothes,Online,C,1/22/2014,397453010,2/20/2014,4033,109.28,35.84,440726.24,144542.72,296183.52


# 9. Conclusion

The dataset has been successfully loaded and explored at a structural level.

The exploration confirms the dataset dimensions, available variables, and data types. Initial descriptive statistics have also been generated, providing a basic understanding of the numerical variables.

The next step is **Data Profiling**, where the quality of the dataset will be evaluated by examining missing values, duplicate records, unique values, and other potential data quality issues.